loops/egn_register.cpp

```diff
@@ -0,0 +1,198 @@
+#include <iostream>
+#include <cmath>
+
+using namespace std;
+
+int main() {
+    long long egn, temp;
+    short int num_count = 0, ctrl_num, sex, region, dd, mm, yy, reg_min_num, valid_check = 0, pow = 512;
+    cin >> egn;
+    temp = egn;
+    ctrl_num = temp % 10;
+    temp = temp / 10;
+    sex = temp % 10;
+    bool isFemale = sex % 2 != 0;
+    region = temp % 10 + temp / 10 % 10 * 10 + temp / 100 % 10 * 100;
+    temp = temp / 1000;
+    dd = temp % 10 + temp / 10 % 10 * 10;
+    temp = temp / 100;
+    mm = temp % 10 + temp / 10 % 10 * 10;
+    temp = temp / 100;
+    yy = temp % 10 + temp / 10 % 10 * 10;
```

Comment:
Може да се сметне далеч по-елегантно: 
```suggestion
    ctrl_num = temp % 10;
    temp = temp / 10;
    sex = temp % 10;
    bool isFemale = sex % 2;  //няма нужда от проверка дали е !=0, резултатът е същия
    region = temp % 1000;
    temp  /= 1000;
    dd = temp % 100;
    temp /= 100;
    mm = temp % 100;
    temp  /= 100;
    yy = temp;
```